### DISCUSSION TITLE
Changed how husk transformations are handled. Please undo.

### DISCUSSION THREAD

**Discussion by SydWad** [FakeFishGames/Barotrauma#10012] [Unstable feedback]
In the recent updates alien infection afflictions were changed from default "[speciesname]husk" to inferring the [speciesname] part and just adding the husk species name/group.

Im not sure why this was done. My issue is i was previously able to turn any species into this husk and it was especially useful for say _"turning a human and a humanhusk into moddedhusk"_ with an infection. Aka having multiple creatures turn into the same creature after the infection. Having mudraptors, crawlers and threshers turn into the same mutated abomination.
This change really limits what kind of transformations you can do with these infections and how dynamic you can make them.

Is there another reason this was done that im missing?
I would really like this change reverted as from my point of view it was a shortsighted attempt to simplify a part of XML modding.

**Comment by SydWad** [comment]
The old way could also be used to turn a human into a monster than back into a human again without loosing their skills.

**Comment by itchyOwl** [comment]
Sorry for the late response. We've registered your feedback. It was changed, because there were issues with the old implementation, making the custom husk afflictions _that didn't override the vanilla affliction_ not (always) work correctly. Can you upload your modded files here or drop a link to the mod?

**Reply by SydWad** [reply]
I no longer have the files for my mod because i had to overhaul it after the update, its a lot more convoluted to have multiple husk transformations now.

I can create an example that used to work and post it here if you like. Hopefully that would help.
If its possible to have the change reverted or revised that would be amazing.

**Reply by SydWad** [reply]
I hope this helps explain my predicament.

![Change_In_Husk_Infection](https://user-images.githubusercontent.com/106110714/205406959-6ed5cd73-f22a-48d8-9059-9ef50481eb9b.png)

In order to maintain info and/or items i have to make a duplicate of each type.
